### PR TITLE
Implement Filecoin Flavor

### PR DESCRIPTION
**Pull request by seesemichaelj**
Per the request of @tcoulter, this PR targets all of the changes in `ganache` in a single PR. All of the intermediate branches still exist in case someone else wants them

### CODE REVIEW

File: src/main/index.js [hunk 1]
```diff
@@ -75,6 +74,10 @@ if (isDevMode) {
   app.getVersion = () => version;
 }
 
+if (isDevelopment) {
```
Comment (tcoulter) [inline src/main/index.js:77]: Can you add a comment above this describing what it's used for?

File: src/main/index.js [hunk 2]
```diff
@@ -330,6 +333,12 @@ app.on('ready', () => {
     mainWindow.focus();
     mainWindow.setTitle("Ganache");
 
+    if (isDevelopment) {
```
Comment (tcoulter) [inline src/main/index.js:336]: Same thing here?

File: src/renderer/init/Core.js [hunk 3]
```diff
@@ -110,6 +159,27 @@ export function initCore(store) {
...
+    store.dispatch(setCurrentOptions(data.options));
+  });
+
+  ipcRenderer.on(SET_STORAGE_DEAL_STATUS_ENUM, (event, data) => {
+    store.dispatch(setStorageDealStatusEnum(data.StorageDealStatus));
+  });
+
   ipcRenderer.on(SHOW_HOME_SCREEN, () => {
     store.dispatch(showHomeScreen());
   });
```
Comment (tcoulter) [inline src/renderer/init/Core.js]: Can you separate these into sections: Common, Ethereum, Filecoin? With very large comment separators. :)

**Review by tcoulter — CHANGES_REQUESTED** [review]
Nice! Few requests:

**Review by tcoulter — APPROVED** [review]
Looks good!

**Review by davidmurdoch — APPROVED** [review]

**Review by davidmurdoch — APPROVED** [review]

**Pull request merged**